### PR TITLE
remove show(hide) editor button

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/50026

Removes "Show Editor" button from the qb header

also fixes segment control and shows 3 buttons for a table and pivot table

https://github.com/user-attachments/assets/2ddb5d54-9f75-4a2d-939b-df3919ac1dcb

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js [hunk 1]
```diff
@@ -1122,7 +1122,9 @@ describe("scenarios > visualizations > pivot tables", { tags: "@slow" }, () => {
     });
   });
 
-  it("should not have to wait for data to show fields in summarisation (metabase#26467)", () => {
+  // There is no way to go to notebook editor while the question result is not loaded anymore,
+  // so we either need to show editor control anyway or remove this test
+  it.skip("should not have to wait for data to show fields in summarization (metabase#26467)", () => {
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js:1127]: in current implementation, we do not render footer before the result is loaded, but we also removed the button from `ViewTitleHeader`, that's why there is no way to return to the notebook editor through UI.

Probably it's more a design question.

as for `ViewTitleHeader.unit.spec`, we do not render button there anymore, it's a [design decision](https://metaboat.slack.com/archives/C0645JP1W81/p1729514301211119?thread_ts=1729287596.325869&cid=C0645JP1W81)

File: frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx [hunk 2]
```diff
@@ -90,19 +96,24 @@ export const LeftViewFooterButtonGroup = ({
             </Tooltip>
           ),
         },
-      ].filter(isNotNull),
-    [dispatch, isNative, vizIcon],
+      ].filter(Boolean),
+    [dispatch, isVisualized, shouldShowEditorButton, vizIcon],
   );
 
+  function getSelectedControlValue() {
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx:103]: good point, we do not render this component at all when there is no `result`, but maybe we should also check `result.error`

**Review by romeovs — APPROVED** [review]
Other than Kamil's comments, this looks good!

File: e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ import {
   filterWidget,
   getNotebookStep,
   modal,
+  notebookButton,
```
Comment (kamilmielnik) [inline e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:20]: 1. New > Question > Orders > Save > Visualize

:x: The "notebook" button is active
:question: the results/visualization button is not shown

image

Comment (uladzimirdev) [reply e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:20]: that's correct. same behavior you'll see on stats. but active state is probably not something we'd like to see, unfortunately segment control marks the item active even if I don't pass any value to it.

I'll double check with Alessio

Comment (AlessioLaiso) [reply e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:20]: Is this a visualization whose chart type is table? Is so, the 3 segments should be shown and this should be the viz, not the editor. Let me know if I'm interpreting this wrong.

Comment (uladzimirdev) [reply e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:20]: yes, it's a table chart type and in master we do not show controls to switch between viz and tabular data for a table, that's why it's implemented like this

https://github.com/user-attachments/assets/a55666d8-013a-4a0a-bca6-615f744eef14

I don't mind if we show all 3 segments anyway

Comment (AlessioLaiso) [reply e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:20]: I see. Not switching between results table and viz table can make sense, but let's at least show two segments for the editor + table.

File: e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js:1127]: @kamilmielnik accordingly with feedback this test still makes sense and I'll fix it in a followup PR, now I'm going to unskip it

File: frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx:106]: we decided to show all buttons in all cases

**Pull request merged**